pkg/api/dirs.go

```diff
@@ -182,8 +181,8 @@ func storeDir(ctx context.Context, reader io.ReadCloser, s storage.Storer, mode
 		return swarm.ZeroAddress, fmt.Errorf("entry marshal: %w", err)
 	}
 
-	sp = splitter.NewSimpleSplitter(s, mode)
-	manifestFileReference, err := file.SplitWriteAll(ctx, sp, bytes.NewReader(fileEntryBytes), int64(len(fileEntryBytes)), toEncrypt)
+	pipe = pipeline.NewPipeline(ctx, s, mode)
+	manifestFileReference, err := pipeline.FeedPipeline(ctx, pipe, bytes.NewReader(fileEntryBytes), int64(len(fileEntryBytes)))
```

Comment:
Same as the above as it appears it is not used anywhwre else.

Reply:
but it reuses the same variable so hopefully less allocations

Reply:
Probably the same, as it will do the copy in the constructor which will happen in both cases. Not sure, though :)